### PR TITLE
Feature/seo schemaorg book microdata

### PR DESCRIPTION
**Pull request by mekarpeles**

### CODE REVIEW

File: openlibrary/templates/books/show.html [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+              $ authorlist = ', '.join(['<span itemprop="author">%s</span>' % thingrepr(author).rstrip() for author in book.authors])
             $:authorlist
             </span>
         <span class="resultPublisher">
             $if book.publishers:
-                $(', '.join(book.publishers)), $book.publish_date
+                $(', '.join(['<span itemprop="publisher">%s</span>' % publisher for publisher in book.publishers])), <span itemprop="datePublished">$book.publish_date</span>
             $else:
-                $book.publish_date
+                <span c$book.publish_date
```
Comment (bfalling) [inline openlibrary/templates/books/show.html]: Is this change correct? `<span c$book.publish_date`?

**Comment by bfalling** [comment]
I had one code question in `openlibrary/templates/books/show.html`, comment left inline.

Were automated tests run for each of the page types affected? I've been putting integration tests under `tests/integration`, and even though they can't run on CI, I run them manually before each merge request. (I'd love to be able to run them headless in CI too, but haven't gotten there yet.)

File: openlibrary/templates/books/carousel_item.html [hunk 2]
```diff
@@ -8,15 +8,16 @@
     $ long_title = truncate(book.title, 40)
 
 <div class="coverMagic hiddenCoverMagic">
+  <span itemscope itemtype="http://schema.org/Book">
```
Comment (LeadSongDog) [inline openlibrary/templates/books/carousel_item.html]: This could/should be https vice http

File: openlibrary/templates/books/show.html [hunk 3]
```diff
@@ -1,22 +1,23 @@
 $def with (book)
 
-<span class="bookcover">
+<span itemscope itemtype="http://schema.org/Book">
```
Comment (LeadSongDog) [inline openlibrary/templates/books/show.html]: This could/should be https

File: openlibrary/templates/work_search.html [hunk 4]
```diff
@@ -233,28 +233,28 @@
             $ doc = get_doc(result_doc)
             $ key = doc.key
             $ work_keys.append(key)
-            <li>
+            <li itemscope itemtype="http://schema.org/Book">
```
Comment (LeadSongDog) [inline openlibrary/templates/work_search.html]: This could/should be https

File: openlibrary/templates/books/works-show.html [hunk 5]
```diff
@@ -1,13 +1,14 @@
 $def with (book)
 
-<span class="bookcover">
+<span itemscope itemtype="http://schema.org/Book">
```
Comment (LeadSongDog) [inline openlibrary/templates/books/works-show.html]: This could/should be https

**Pull request merged**